packages/powersync_sqlcipher/lib/src/sqlcipher.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:powersync_core/powersync_core.dart';
+import 'package:powersync_core/sqlite3_common.dart';
+import 'package:powersync_core/sqlite3_open.dart' as sqlite3_open;
+import 'package:powersync_core/sqlite_async.dart';
+import 'package:sqlcipher_flutter_libs/sqlcipher_flutter_libs.dart';
+
+/// A factory for opening a database with SQLCipher encryption.
+/// An encryption [key] is required to open the database.
+class PowerSyncSQLCipherOpenFactory extends PowerSyncOpenFactory {
+  PowerSyncSQLCipherOpenFactory(
+      {required super.path, required this.key, super.sqliteOptions});
+
+  final String key;
+
+  @override
+  List<String> pragmaStatements(SqliteOpenOptions options) {
+    final basePragmaStatements = super.pragmaStatements(options);
+    return [
+      // Set the encryption key as the first statement
+      "PRAGMA KEY = '$key'",
```

Comment:
```suggestion
      "PRAGMA KEY = ${quoteString(key)}",
```

This makes sure there is no SQL injection possibility if user input is directly used as the key. The `quoteString` function is part of sqlite_async.